lib/search_elastic/index_create.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+class SearchElasticIndexCreator
+  def initialize(service_name:, index_name:, shards:, replicas:)
+    @client = nil
+    @service_name = service_name.upcase
+    raise ArgumentError, \
+      "service_name must be 'ELASTICSEARCH' or 'OPENSEARCH'" \
+      unless %w[ELASTICSEARCH OPENSEARCH].include?(@service_name)
+
+    @index_name = index_name
+    @shards = shards
+    @replicas = replicas
+  end
+
+  def create_or_update_index(client)
+    @client = client
+    index_pattern = "*#{@index_name}*"
+    template_generator = SearchElastic::Template.new(index_pattern)
+    template_body = JSON.parse(template_generator.body)
```

Comment:
Can you add a comment why this is now necessary?, It seems that it's for setting the shards, and replicas but I don't see it being set before on the template, can we pass this parameters to the template directly?